config_files/config_example_coca.yaml

```diff
@@ -0,0 +1,206 @@
+modalities_setup:
+  run_mode: FROM_SCRATCH
+  settings:
+    global_num_seen_samples: 0
+
+wandb:
+  project_name: modalities
+  mode: OFFLINE
+
+data:
+  sample_key: "input_ids"
+  target_key: "target_ids"
+  sequence_len: 128
+  train_dataloader:
+    type_hint: LLMDataLoader
+    config:
+      num_workers: 2
+      pin_memory: true
+      shuffle: false
+      dataloader_tag: "train"
+      dataset:
+        type_hint: DummyDataset
+        config:
+          num_samples: 4
+          sample_definition:
+            - - images
+              - [3, 224, 224]
+              - float
+            - - input_ids
+              - [1024]
+              - int
+      batch_sampler:
+        type_hint: BatchSampler
+        config:
+          batch_size: 3
+          drop_last: false
+          sampler:
+            type_hint: DistributedSampler
+            config:
+              rank: ${training.global_rank}
+              num_replicas: ${training.world_size}
+              shuffle: true
+      collate_fn:
+        type_hint: CoCaCollator
+        config:
+          sample_keys:
+            - "images"
+            - ${data.sample_key}
+          target_keys: []
+          text_sample_key: ${data.sample_key}
+          text_target_key: ${data.target_key}
+  eval_dataloaders:
+    - type_hint: LLMDataLoader
+      config:
+        num_workers: 2
+        pin_memory: true
+        shuffle: false
+        dataloader_tag: "val"
+        dataset: ${data.train_dataloader.config.dataset}
+        batch_sampler:
+          type_hint: BatchSampler
+          config:
+            batch_size: 3
+            drop_last: false
+            sampler:
+              type_hint: DistributedSampler
+              config:
+                rank: ${training.global_rank}
+                num_replicas: ${training.world_size}
+                shuffle: true
+        collate_fn: ${data.train_dataloader.config.collate_fn}
+    - type_hint: LLMDataLoader
+      config:
+        num_workers: 2
+        pin_memory: true
+        shuffle: false
+        dataloader_tag: "test"
+        dataset: ${data.train_dataloader.config.dataset}
```

Comment:
Linking the train data in val and test might encourage bad practices (even if it's just a dummy dataset right now).